spec/app/models/metasploit/cache/module/path_spec.rb

```diff
@@ -643,14 +651,216 @@
+          end
+        end
+
+        context 'without same Metasploit::Cache::Module::Path for name and real path collision' do
+          #
+          # lets
+          #
+
+          let(:real_path) {
+            real_path_collision.real_path
+          }
+
+          #
+          # let!s
+          #
+
+          let!(:real_path_collision) {
+            FactoryGirl.create :metasploit_cache_module_path
+          }
+
+          specify {
+            expect {
+              resolve_collisions
+            }.to raise_error(
+                     ActiveRecord::RecordNotUnique,
+                     "Collision against two pre-existing #{name.pluralize}: (1) on gem (#{name_collision.gem}) and " \
+                     "name (#{name_collision.name}) and (2) on real_path (#{real_path_collision.real_path})."
+                 )
+          }
+        end
+      end
+
+      context 'without real path collision' do
+        let(:real_path) {
+          FactoryGirl.generate :metasploit_cache_module_path_real_path
+        }
+
+        it 'updates #real_path on name collision to real_path' do
+          expect {
+            resolve_collisions
+          }.to change {
+                 name_collision.reload.real_path
+               }.to real_path
+        end
+
+        it 'returns name collision' do
+          expect(resolve_collisions).to eq(name_collision)
+        end
+      end
+    end
+
+    context 'without name collision' do
+      context 'with real path collision' do
+        #
+        # lets
+        #
+
+        let(:real_path) {
+          real_path_collision.real_path
+        }
+
+        let(:real_path_collision) {
+          FactoryGirl.create :named_metasploit_cache_module_path
+        }
+
+        context 'with named' do
+          let(:gem) {
+            FactoryGirl.generate :metasploit_cache_module_path_gem
+          }
+
+          let(:name) {
+            FactoryGirl.generate :metasploit_cache_module_path_name
+          }
+
+          it 'updates #gem on real path collision to gem' do
+            expect {
+              resolve_collisions
+            }.to change {
+                   real_path_collision.reload.gem
+                 }.to gem
+          end
+
+          it 'updates #name on real path collision to name' do
+            expect {
+              resolve_collisions
+            }.to change {
+                   real_path_collision.reload.name
+                 }.to name
+          end
+
+          it 'returns real path collision' do
+            expect(resolve_collisions).to eq(real_path_collision)
+          end
+        end
+
+        context 'without named' do
+          let(:gem) {
+            nil
+          }
+
+          let(:name) {
+            nil
+          }
+
+          it 'does not erase #gem on real path collision' do
+            expect {
+              resolve_collisions
+            }.not_to change {
+                   real_path_collision.reload.gem
+                 }
```

Comment:
`end` at 809, 17 is not aligned with `change {` at 807, 21 or `}.not_to change {` at 807, 12.